dwave_networkx/algorithms/exact_cover.py

```diff
@@ -0,0 +1,174 @@
+    cover of the problem set, if it exists, and uses the sampler to sample from
+    it.
+
+    Parameters
+    ----------
+    problem_set : iterable
+        An iterable of unique numbers
+
+    subsets : list(iterable(numeric))
+        A list of subsets of `problem_set` used to find an exact cover.
+
+    sampler
+        A binary quadratic model sampler. A sampler is a process that samples
+        from low energy states in models defined by an Ising equation or a
+        Quadratic Unconstrained Binary Optimization Problem (QUBO). A sampler is
+        expected to have a 'sample_qubo' and 'sample_ising' method. A sampler is
+        expected to return an iterable of samples, in order of increasing
+        energy. If no sampler is provided, one must be provided using the
+        `set_default_sampler` function.
+
+    sampler_args
+        Additional keyword parameters are passed to the sampler.
+
+    Returns
+    -------
+    exact_cover : list(iterable(numeric))/None
+       List of subsets that form an exact cover, or None if an exact cover does
+       not exist.
+
+    Examples
+    --------
+    This example uses a sampler from `dimod
+    <https://github.com/dwavesystems/dimod>`_ to find an exact cover of the set
+    {1, 2, 3, 4, 5, 6}, given a list of some of its subsets.
+
+    >>> import dwave_networkx as dnx
+    >>> import dimod
+    >>> sampler = dimod.ExactSolver()  # small testing sampler
+    >>> problem_set = {1, 2, 3, 4, 5, 6}
+    >>> subsets = [{1, 2, 3}, {1, 4, 5}, {5}, {4, 6}, {6}]
+    >>> dnx.exact_cover(problem_set, subsets, sampler)
+    [{1, 2, 3}, {5}, {4, 6}]
+
+    References
+    ----------
+    Based on the formulation presented in [AL]_
+
+    .. [AL] Lucas, A. (2014). Ising formulations of many NP problems.
+    Frontiers in Physics, Volume 2, Article 5.
+
+    """
+    if not problem_set:
+        raise ValueError("problem_set should not be empty")
+
+    if not subsets:
+        return None
+
+    # Check that problem set is valid
+    orig_len = len(problem_set)
+    problem_set = set(problem_set)
+    if len(problem_set) != orig_len:
+        raise ValueError("problem_set should not contain any duplicates")
+
+    # Create BQM
+    bqm = BinaryQuadraticModel({}, {}, 0, 'BINARY')
+
+    offset = 0
+    for element in problem_set:
+        offset += 1
+
+        for i in range(len(subsets)):      
+            if element in subsets[i]:
+                bqm.add_variable(i, -1)
+
+                for j in range(i-1, -1, -1):
+                    if element in subsets[j]:
+                        bqm.add_interaction(i, j, 2)
+
+    bqm.offset = offset
+    
+    response = sampler.sample(bqm)
+
+    # Get lowest energy sample
+    sample = next(iter(response))
+
+    result = []
+    for subset_index in sample:
+        if sample[subset_index] > 0:
+            result.append(subsets[subset_index])
+
+    # Validate that result is actually an exact cover
```

Comment:
Most of the other dnx functions return the solution whether it is correct or not, but document it.